core/src/main/scala/monocle/std/Set.scala

```diff
@@ -11,7 +11,7 @@ trait SetOptics {
     def empty = Prism[Set[A], Unit](s => if(s.isEmpty) Some(()) else None)(_ => Set.empty[A])
   }
 
-  implicit def atSet[A]: At[Set[A], A, Unit] = new At[Set[A], A, Unit] {
+  implicit def atSet[A]: At[Set[A], A, Option[Unit]] = new At[Set[A], A, Option[Unit]] {
```

Comment:
very good point. In fact, I would say it is an argument for not separating `At`.